### PR TITLE
feat: NumberInput component redesigned

### PR DESCRIPTION
**Pull request by khnbzkrt**

### CODE REVIEW

File: examples/fineFoods/client/src/components/numberInput.tsx [hunk 1]
```diff
@@ -10,9 +10,22 @@ export const NumberInput: React.FC<NumberInputProps> = ({
...
+                onClick={() =>
+                    setValue((prev) => {
+                        if (prev > 0) {
+                            return prev - 1;
+                        }
+                        return 0;
+                    })
+                }
+            >
+                <ChevronDownIcon className="text-primary h-full w-4" />
```
Comment (aliemir) [inline examples/fineFoods/client/src/components/numberInput.tsx:25]:
```suggestion
                <ChevronDownIcon className="text-primary h-4 w-4" />
```

Can we also change this? It's breaking on safari 😅

File: examples/fineFoods/client/src/components/numberInput.tsx [hunk 2]
```diff
@@ -22,31 +35,18 @@ export const NumberInput: React.FC<NumberInputProps> = ({
...
+                onClick={() =>
+                    setValue((prev) => {
+                        if (prev >= 0) {
+                            return prev + 1;
+                        }
+                        return 0;
+                    })
+                }
+            >
+                <ChevronUpIcon className="text-primary h-full w-4" />
```
Comment (aliemir) [inline examples/fineFoods/client/src/components/numberInput.tsx:49]:
```suggestion
                <ChevronUpIcon className="text-primary h-4 w-4" />
```

Same as above, breaking the icon in safari.

**Comment by aliemir** [comment]
Broken checks are fixed with #2135, can you please merge `next` to this branch? 🙏

**Comment by khnbzkrt** [comment]
> Broken checks are fixed with #2135, can you please merge `next` to this branch? 🙏

Sorry to late answer. My computer is broken on holiday time. What can i do now for help?

File: examples/fineFoods/client/src/components/numberInput.tsx [hunk 3]
```diff
@@ -10,9 +10,22 @@ export const NumberInput: React.FC<NumberInputProps> = ({
...
+                        }
+                        return 0;
+                    })
+                }
+            >
+                <ChevronDownIcon className="text-primary h-4 w-4" />
+            </button>
             <input
-                className="focus:outline-primary focus:border-primary h-full w-16 rounded-md border py-2 pl-2 transition-colors"
+                className="focus:outline-primary focus:border-primary h-full w-16 border p-2 transition-colors"
```
Comment (omeraplak) [inline examples/fineFoods/client/src/components/numberInput.tsx:28]:
```suggestion
                className="focus:outline-primary focus:border-primary h-full w-8 border p-2 transition-colors"
```

**Comment by omeraplak** [comment]
Hey @khnbzkrt ,
Thank you!

**Pull request merged**